src/awst_build/eb/arc4/base.ts

```diff
@@ -85,3 +101,103 @@ class Arc4EqualsFunctionBuilder extends FunctionBuilder {
     )
   }
 }
+
+export class Arc4EncodedFromBytesFunctionBuilder extends FunctionBuilder {
+  constructor(
+    sourceLocation: SourceLocation,
+    private ptypeFactory?: (args: PType[]) => ARC4EncodedType,
+    private genericArgsCount?: number,
+  ) {
+    super(sourceLocation)
+  }
+  call(args: ReadonlyArray<NodeBuilder>, typeArgs: ReadonlyArray<PType>, sourceLocation: SourceLocation): NodeBuilder {
+    const funcName = 'fromBytes'
+    const {
+      ptypes,
+      args: [initialValueBuilder],
+    } = parseFunctionArgs({
+      args,
+      typeArgs,
+      genericTypeArgs: this.genericArgsCount ?? 1,
+      funcName,
+      argSpec: (a) => [a.required(bytesPType)],
+      callLocation: sourceLocation,
+    })
+
+    const ptype = (this.ptypeFactory ? this.ptypeFactory(ptypes) : ptypes[0]) as ARC4EncodedType
+    const initialValue = initialValueBuilder.resolveToPType(bytesPType).resolve()
+    const initialValueExpr =
+      initialValue instanceof BytesConstant
+        ? nodeFactory.bytesConstant({
+            value: initialValue.value,
+            wtype: ptype.wtypeOrThrow,
+            sourceLocation: sourceLocation,
+          })
+        : nodeFactory.reinterpretCast({
+            wtype: ptype.wtype,
+            sourceLocation,
+            expr: initialValue,
+          })
+
+    return typeRegistry.getInstanceEb(initialValueExpr, ptype)
+  }
+}
+
+export class Arc4EncodedFromLogFunctionBuilder extends FunctionBuilder {
+  constructor(
+    sourceLocation: SourceLocation,
+    private ptypeFactory?: (args: PType[]) => ARC4EncodedType,
+    private genericArgsCount?: number,
+  ) {
+    super(sourceLocation)
+  }
+
+  call(args: ReadonlyArray<NodeBuilder>, typeArgs: ReadonlyArray<PType>, sourceLocation: SourceLocation): NodeBuilder {
+    const funcName = 'fromLog'
+    const {
+      ptypes,
+      args: [initialValueBuilder],
+    } = parseFunctionArgs({
+      args,
+      typeArgs,
+      genericTypeArgs: this.genericArgsCount ?? 1,
+      funcName,
+      argSpec: (a) => [a.required(bytesPType)],
+      callLocation: sourceLocation,
+    })
+
+    const ptype = (this.ptypeFactory ? this.ptypeFactory(ptypes) : ptypes[0]) as ARC4EncodedType
+    const initialValue = initialValueBuilder.resolveToPType(bytesPType).resolve()
```

Comment:
Shouldn't need the extra resolveToPType here, it's already been called by the parseFunctionArgs helper

Reply:
It has been updated in this commit, 6c31fad.